File: leetcode/_contest/bi-weekly-contest-14/2.go
package main

func removeInterval(intervals [][]int, toBeRemoved []int) [][]int {
	res := [][]int{}

	a, b := toBeRemoved[0], toBeRemoved[1]

	for i := range intervals {
		x, y := intervals[i][0], intervals[i][1]

		if x > a && y < b {
			continue
		}

		if a > x && b < y {
			res = append(res, []int{x, a})
			res = append(res, []int{b, y})
			continue
		}

		if a > x && a < y {
			y = a
		}

		if b > x && b < y {
			x = b
		}

		res = append(res, []int{x, y})
	}

	return res
}

File: leetcode/_contest/bi-weekly-contest-14/2.js
/**
 * @param {number[][]} intervals
 * @param {number[]} toBeRemoved
 * @return {number[][]}
 */
var removeInterval = function(intervals, toBeRemoved) {
  const res = [];

  const [ a, b ] = toBeRemoved;
  for (let i = 0; i < intervals.length; i++) {
    let [ x, y ] = intervals[i];

    if (x > a && y < b) {
      continue;
    }

    if (a > x && b < y) {
      res.push([ x, a ]);
      res.push([ b, y ]);
      continue;
    }

    if (a > x && a < y) {
      y = a;
    }

    if (b > x && b < y) {
      x = b;
    }

    res.push([ x, y ]);
  }

  return res;
};

let intervals = [ [ 0, 2 ], [ 3, 4 ], [ 5, 7 ] ],
  toBeRemoved = [ 1, 6 ];

console.log(removeInterval(intervals, toBeRemoved));
(intervals = [ [ 0, 5 ] ]), (toBeRemoved = [ 2, 3 ]);
console.log(removeInterval(intervals, toBeRemoved));
